src/timesync/timesyncd-manager.c

```diff
@@ -976,6 +977,14 @@ static int manager_network_read_link_servers(Manager *m) {
         STRV_FOREACH(i, ntp) {
                 bool found = false;
 
+                r = dns_name_is_valid_or_address(*i);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to check validity of NTP server name or address '%s': %m", *i);
```

Comment:
i don't think this should be fatal. i think it would be better to deal with this gracefully: ignore the entry and proceed.

Reply:
Hmm, as already explained by @keszybz in the above, the negative error is mostly `-ERRNO`, I think. So, in that case, it is hard to continue the following process.

Reply:
BTW, in the negative error case, we should not directly return here. In the revised version I use `goto clear` instead.